prepare-image.sh

```diff
@@ -41,13 +41,23 @@ if [[ "$INSTALL_TYPE" == "source" ]]; then
 
     python3 -c 'import os; import sys; import jinja2; sys.stdout.write(jinja2.Template(sys.stdin.read()).render(env=os.environ, path=os.path))' < "${IRONIC_PKG_LIST}" > "${IRONIC_PKG_LIST_FINAL}"
 
+    UPPER_CONSTRAINTS_PATH="/tmp/${UPPER_CONSTRAINTS_FILE}"
+
+    # NOTE(elfosardo): if the content of the upper-constraints file is empty,
+    # we give as assumed that we're on the master branch
+    if [[ -s "${UPPER_CONSTRAINTS_PATH}" ]]; then
+        curl -L https://releases.openstack.org/constraints/upper/master -o "${UPPER_CONSTRAINTS_PATH}"
+    fi
+
     if [[ -n ${SUSHY_SOURCE:-} ]]; then
-        curl -L "${UPPER_CONSTRAINTS_FILE:-https://releases.openstack.org/constraints/upper/master}" -o /tmp/sushy-constraints.txt
-        UPPER_CONSTRAINTS_FILE="/tmp/sushy-constraints.txt"
-        sed -i '/^sushy===/d' "$UPPER_CONSTRAINTS_FILE"
+        sed -i '/^sushy===/d' "$UPPER_CONSTRAINTS_PATH"
+    fi
+
+    if [[ -n ${IRONIC_LIB_SOURCE:-} ]]; then
+        sed -i '/^python-ironic-lib===/d' "$UPPER_CONSTRAINTS_PATH"
```

Comment:
It's called just ironic-lib: https://github.com/openstack/requirements/blob/master/upper-constraints.txt#L598